utils/data_preprocessing/employee_growth_per_quater_per_domain.py

```diff
@@ -0,0 +1,128 @@
+import sys
+sys.path.append("../..")
+import ast
+import pandas as pd
+from constants import constants
+from utils.exception import *
+from collections import defaultdict
+from constants.schema.emp_growth_per_quater_per_domain_fields import TENANT_ID , DOMAIN_NAME, QUATER, PERCENT_INCREASE
+from domains.new_dev_project.domain_infrastructure.initial_data_impl import InitialData
+
+
+def convert_to_dict(value):
+        try:
+            # Try to convert string representation of a dictionary to a Python dictionary
+            return ast.literal_eval(value)
+        except Exception as e:
+            # Raise custom error if conversion fails
+            raise MyError(error_code=400, error_message="Cannot process data, input file type not supported yet!")
+        
+
+
+# Function to group the employee count by year and month
+def group_by_year(value):
+    # Create a defaultdict to store the counts by year
+    grouped_data = defaultdict(dict)
+    
+    # Iterate through the months and counts in the given dictionary
+    for month_year, count in value.items():
+        # Extract year and month from the key
+        year, month = month_year.split('-')
+        year = int(year)  # Convert year to integer
+        month = int(month)  # Convert month to integer
+        
+        # Group the data by year, with months as keys and counts as values
+        grouped_data[year][month] = count
+    
+    # Convert defaultdict to regular dict for output
+    return dict(grouped_data)
+
+
+
+
+def retrieve_data_from_s3_df_for_employee_growth_per_quater_for_each_domain():
```

Comment:
Not an ideal file name. We do not need s3 in function name. This could just be named as 
retrieve_employee_growth_per_quater . This provides all the required context without being extremely verbose